index.ts

```diff
@@ -140,6 +140,53 @@ const mem = <
 
 export = mem;
 
+/**
+Returns a decorator which memoizes the function provided.
+
+@example
+```
+import mem = require('mem');
+
+class Example {
+	constructor() {
+		this.i = 0;
+	}
+
+	@mem.decorator()
+	counter() {
+		return ++this.i;
+	}
+}
+
+class ExampleWithOptions {
+	constructor() {
+		this.i = 0;
+	}
+
+	@mem.decorator({maxAge: 1000})
+	counter() {
+		return ++this.i;
+	}
+}
+```
+*/
+mem.decorator = <
+	FunctionToMemoize extends AnyFunction,
+	CacheKeyType
+>(
+	options: Options<FunctionToMemoize, CacheKeyType> = {}
+) => (
+	target: any,
+	propertyKey: string,
+	descriptor: PropertyDescriptor
+): void => {
+	if (typeof target[propertyKey] !== 'function') {
```

Comment:
Would it make sense to assign `target[propertyKey]` to a seperate variable to prevent double lookup or is that just a pointless microoptimisation?